kernel/solo5.h

```diff
@@ -136,8 +165,13 @@ uint64_t solo5_clock_monotonic(void);
 uint64_t solo5_clock_wall(void);
 
 /*
- * Blocks until monotonic time reaches until_nsecs or I/O is possible,
- * whichever is sooner. Returns 1 if I/O is possible, otherwise 0.
+ * Blocks until monotonic time reaches until_nsecs or network I/O is possible,
+ * whichever is sooner.
+ *
+ * If network I/O is possible:
+ *     Returns 1.
+ * If monotonic time has reached until_nsecs and network I/O is not possible:
+ *     Returns 0.
  *
  * TODO: Extend this interface to select which I/O events are of interest.
  */
```

Comment:
To be more like the POSIX poll(), the time should be a delta and not a future timestamp. Plus, that would remove this time reading in the mirage event loop:

```
        let timeout =
          match Time.select_next () with
          |None -> Time.Monotonic.(time () + of_nanoseconds 86_400_000_000_000L) (* one day = 24 * 60 * 60 s *)
          |Some tm -> tm
        in
        if poll timeout then begin
```

https://github.com/mirage/mirage-solo5/blob/master/lib/main.ml#L68

On the other hand, that's a risky change, so up to you.